packages/backend/src/core/DiscoveryWatcher.ts

```diff
@@ -69,19 +75,42 @@ export class DiscoveryWatcher {
     this.logger.info('Update finished', { blockNumber })
   }
 
-  async compareWithCommitted(
+  async findChanges(
     name: string,
+    blockNumber: number,
+    timestamp: UnixTime,
     discovered: AnalyzedData[],
     overrides?: Record<string, DiscoveryContract>,
   ): Promise<DiscoveryDiff[]> {
-    const committed = await this.configReader.readDiscovery(name)
     const parsedDiscovery = parseDiscoveryOutput(discovered)
+    const committed = await this.configReader.readDiscovery(name)
+    const databaseEntry = await this.repository.getLatest(name)
 
-    return diffDiscovery(
-      committed.contracts,
-      parsedDiscovery.contracts,
-      overrides ?? {},
-    )
+    let diff = []
+    if (databaseEntry.length === 0) {
+      diff = diffDiscovery(
+        committed.contracts,
+        parsedDiscovery.contracts,
+        overrides ?? {},
+      )
+    } else {
+      diff = diffDiscovery(
+        databaseEntry[0].discovery.contracts,
+        parsedDiscovery.contracts,
+        overrides ?? {},
+      )
+    }
```

Comment:
```suggestion
    const currentContracts = databaseEntry.length === 0 ? committed.contracts : databaseEntry[0].discovery.contracts
    const diff = diffDiscovery(
      currentContracts,
      parsedDiscovery.contracts,
      overrides ?? {},
    )
```

Would be even better if you could get currentContracts from a separate method